### PR TITLE
Pass buf into _convert_err_msg and remove global variable

### LINKED ISSUE #89: Fix undefined `buf` in `_convert_err_msg`
## Dev Effort
0.5D

## Description
`_convert_err_msg` prints `buf.getvalue()` even though `buf` isn't passed in as a parameter and isn't (usually) a global.

Based on 5a2d7517738649e6a2847857f2058b8f4211d2ba the intention was that `buf` be a global. Since all the other parameters are explicitly passed in, buf should be as well to reduce confusion.

### PR DESCRIPTION
**Pull request by ablwr**
Resolve #89

### CODE REVIEW

File: fido/prepare.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def doByte(chars, i, littleendian, esc=True):
     c1 = '0123456789ABCDEF'.find(chars[i].upper())
     c2 = '0123456789ABCDEF'.find(chars[i + 1].upper())
     if (c1 < 0 or c2 < 0):
-        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars))
+        raise Exception(_convert_err_msg('bad byte sequence', chars[i:i + 2], i, chars, buf=None))
```
Comment (carlwilson) [inline fido/prepare.py]: Is this tested? I have a horrible feeling that the `buf.getValue()` in `_convert_err_msg` will crash when `buf=None`. Something (very) like `AttributeError: 'NoneType' object has no attribute 'getvalue'`

Comment (ablwr) [reply fido/prepare.py]: You're right -- sorry I should have put a [WIP] flag on this!!

Comment (carlwilson) [reply fido/prepare.py]: No problem

**Comment by ablwr** [comment]
Okay, I rebased and just added a new `buf` here. I did some testing and there wasn't a crash when hitting this function. I'm not saying this is the best solution, but it is a solution.

**Pull request merged**